modules/axfs/src/fs/fatfs.rs

```diff
@@ -0,0 +1,262 @@
+                Ok(())
+            }
+            _ => Err(VfsError::Unsupported),
+        }
+    }
+
+    fn remove(&self, path: &str) -> VfsResult {
+        debug!("remove at fatfs: {}", path);
+        let path = path.trim_matches('/');
+        assert!(!path.is_empty()); // already check at `root.rs`
+        if let Some(rest) = path.strip_prefix("./") {
+            return self.remove(rest);
+        }
+        self.0.remove(path).map_err(as_vfs_err)
+    }
+
+    fn read_dir(&self, start_idx: usize, dirents: &mut [VfsDirEntry]) -> VfsResult<usize> {
+        let mut iter = self.0.iter().skip(start_idx);
+        for (i, out_entry) in dirents.iter_mut().enumerate() {
+            let x = iter.next();
+            match x {
+                Some(Ok(entry)) => {
+                    let ty = if entry.is_dir() {
+                        VfsNodeType::Dir
+                    } else if entry.is_file() {
+                        VfsNodeType::File
+                    } else {
+                        unreachable!()
+                    };
+                    *out_entry = VfsDirEntry::new(&entry.file_name(), ty);
+                }
+                _ => return Ok(i),
+            }
+        }
+        Ok(dirents.len())
+    }
+}
+
+impl VfsOps for FatFileSystem {
+    fn root_dir(&self) -> VfsNodeRef {
+        let root_dir = unsafe { (*self.root_dir.get()).as_ref().unwrap() };
+        root_dir.clone()
+    }
+}
+
+impl fatfs::IoBase for Disk {
+    type Error = ();
+}
+
+impl Read for Disk {
+    fn read(&mut self, mut buf: &mut [u8]) -> Result<usize, Self::Error> {
+        let mut read_len = 0;
+        while !buf.is_empty() {
+            match self.read_one(buf) {
+                Ok(0) => break,
+                Ok(n) => {
+                    let tmp = buf;
+                    buf = &mut tmp[n..];
+                    read_len += n;
+                }
+                Err(_) => return Err(()),
+            }
+        }
+        Ok(read_len)
+    }
+}
+
+impl Write for Disk {
+    fn write(&mut self, mut buf: &[u8]) -> Result<usize, Self::Error> {
+        let mut write_len = 0;
+        while !buf.is_empty() {
+            match self.write_one(buf) {
+                Ok(0) => break,
+                Ok(n) => {
+                    buf = &buf[n..];
+                    write_len += n;
+                }
+                Err(_) => return Err(()),
+            }
+        }
+        Ok(write_len)
+    }
+    fn flush(&mut self) -> Result<(), Self::Error> {
+        Ok(())
+    }
+}
+
+impl Seek for Disk {
+    fn seek(&mut self, pos: SeekFrom) -> Result<u64, Self::Error> {
+        match pos {
+            SeekFrom::Start(pos) => {
+                self.set_position(pos);
+                Ok(pos)
+            }
+            SeekFrom::Current(off) => {
+                let pos = (self.position() as i64 + off) as u64;
+                self.set_position(pos);
+                Ok(pos)
+            }
```

Comment:
Such as SeekFrom::Current(position: isize), Is it possible that it is less than 0?